BasicSteps/ProcessStep.cs

```diff
@@ -73,9 +78,27 @@ public Int32 Timeout
         ManualResetEvent outputWaitHandle, errorWaitHandle;
         StringBuilder output;
 
+        public ProcessStep()
+        {
+            Rules.Add(HasNoDuplicateEnvironmentVariables, "Cannot have multiple environment variables with the same name.", nameof(EnvironmentVariables));
```

Comment:
Are there some names which are invalid for environment variables? Maybe we should check for those.
Maybe we need some rules like these?
- ```string.IsNullOrWhiteSpace(key)```
- ```key.Trim() == key```

Reply:
I will add the environment variables rules from this page:
Along with what you suggested.
https://docs.microsoft.com/en-us/windows/win32/procthread/environment-variables

Other than that i think we should be really careful about validating these too much, as different platforms might very well have different possibilities for what environment variables they allow or dont allow.